format/format.go

```diff
@@ -1108,13 +1108,29 @@ func (w *writer) objectWriter() entryWriter {
 		entry := x.([2]*ast.Term)
 		comments = w.writeTerm(entry[0], comments)
```

Comment:
What about when the union is in the object key?
E.g.:
```rego
union_object := {(declare3 | declare4): "foo"}
```

Reply:
Updated to handle this case.